telebot/modules/regex.py

```diff
@@ -0,0 +1,65 @@
+from re import compile, search
+from subprocess import Popen, PIPE, check_output, CalledProcessError, STDOUT
+
+from telegram import MAX_MESSAGE_LENGTH, Update
+from telegram.ext import CallbackContext, MessageHandler, Filters
+
+from telebot import dispatcher
+from telebot.functions import bot_action
+from telebot.modules.db.exceptions import get_exceptions_for_chat
+
+
+@bot_action("regex")
+def regex(update: Update, context: CallbackContext):
+    """
+    Replace text using sed and regex
+    :param update: object representing the incoming update.
+    :param context: object containing data about the command call.
+    """
+    if "regex" in get_exceptions_for_chat(update.effective_chat.id):
+        return
+
+    if not update.effective_message.reply_to_message:
+        update.effective_message.reply_text("Gimme text to replace stuff in, baka!")
+        return
+
+    # make sure `s` command is properly terminated
+    command = update.effective_message.text
+    delimiter = command[1]
+    if command.count(delimiter) == 2:
```

Comment:
And also add a check for !=3, assuming you can safely escape other occurences not in the expression

Reply:
can you elaborate a bit more?

Reply:
Basically counting the number of delimiters, to predict an error beforehand itself

Reply:
but what would a !=3 check add to it? we need there to be 3 delimiters for it to be valid, and the regex will match one with minimum 2 delimiters, hence the check for 2 delimiters. Plus in the stage after this I remove everything except for flags after third delimiter, so cases like someone passing 4 delimiters etc. are also handled in that everything after the flags is discarded

Reply:
Hmm, basically I just wanted to try to detect errors in it pehle se, instead of sed returning errors

Reply:
Otherwise, seems fine

Reply:
I thought sed returning errors would be the easiest way of getting proper error messages to user and tell them exactly what's wrong :laughing:

Reply:
In that way, yep, I was thinking of more possible vulnerabilities through that though xD

Reply:
> Otherwise, seems fine

approve then, I wanna merge.....